src/main/kotlin/com/project/hotel/handler/UserHandler.kt

```diff
@@ -0,0 +1,36 @@
+package com.project.hotel.handler
+
+import com.project.hotel.model.users.User
+import com.project.hotel.service.UserService
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.reactive.awaitFirst
+import org.springframework.http.codec.multipart.FilePart
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.server.*
+import org.springframework.web.reactive.function.server.ServerResponse.notFound
+import org.springframework.web.reactive.function.server.ServerResponse.ok
+
+
+@Component
+class UserHandler(private val userService: UserService) {
+    @FlowPreview
+    suspend fun addUser(request: ServerRequest): ServerResponse {
+        var user=request.awaitBody<User>()
+        user= userService.addUser(user)!!
+        return when {
```

Comment:
не конечно пиздато что ты осваиваешь такие вещи как when но нужно и понимать где их использовать 

в даном случае достаточно 
```suggestion
        return if(user != null) { ok().json().bodyValueAndAwait(user) } else { ...  }
```